doc/embedded/crosscompile_example.md

```diff
@@ -0,0 +1,132 @@
+
+Then we need to call `fetch_mlpack()` will download mlpack including all dependencies,
+cross-compile openblas and set up all the necessary parameters to find these dependencies.
+Most of mlpack dependencies are header-only with the exception of Openblas,
+thus this is expected to be a quick step.
+
+`fetch_mlpack()` will detect if cross compilation is necessary or not depending
+on the command that is executed when running cmake. Based on this, it will
+compile OpenBLAS.
+
+```cmake
+cmake_minimum_required(VERSION 3.11)
+project(main)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/CMake")
+include(CMake/Autodownload.cmake)
+include(CMake/ConfigureCrossCompile.cmake)
+
+fetch_mlpack()
+
+```
+
+#### Setting up include directories and source files
+
+The last part of our CMake file consists of merging all of the above components
+together. First we are going to find the OpenMP package if the user defined the
+variable above, then we will start including the directories of our
+program. You will need to do the same for your software if you are trying to
+integrate this example into an existing development, this should be done by
+defining an include variable using `set` directive or add them directly in
+`target_include_directories`.
+Regarding the source code, a similar process by either adding then to
+`add_executables` or by appending the source files to `SOURCES_FILES` variable. 
+
+Finally do not forget to add any external library that you need to link against
+in `target_link_libraries`.
+
+```cmake
+include_directories(BEFORE ${MLPACK_INCLUDE_DIRS})
+include_directories(BEFORE ${CMAKE_CURRENT_SOURCE_DIR}/src/)
+
+## User includes go here
+
+# Finally, add any cross-compilation support libraries (they may need to come
+# last).  If we are not cross-compiling, no changes will happen here.
+set(MLPACK_LIBRARIES ${MLPACK_LIBRARIES} ${CROSS_COMPILE_SUPPORT_LIBRARIES})
+
+## Add your source files to SOURCES_FILES list
+
+add_executable(RandomForest main.cpp ${SOURCES_FILES})
+target_sources(RandomForest PRIVATE ${SOURCE_FILES})
+
+## Do not forget to add the include variables in here.
+
+target_include_directories(RandomForest PRIVATE
+  ${MLPACK_INCLUDE_DIRS}
+  /path/to/your/own/include
+)
+
+## Do not forget to add libraries that you are linking against in here.
+
+target_link_libraries(RandomForest PRIVATE -static
+  ${MLPACK_LIBRARIES}
+)
+```
+
+#### Optimization and crosscompilation
+
+If you are interested in adding specific compiler flags to optimize operations
+on your hardware. Please feel free to look at mlpack
+`CMake/crosscompile-arch-flags.cmake` you can either integrate the variable by
+copy and paste them directly to this CMake file or you can
+copy the entire `crosscompile-arch-config.cmake` file from `mlpack/CMake` and included
+it close to `Autodownload.cmake` at start of this tutorial but in this case you need
+to specify the `ARCH_NAME` variable as we did in the this [tutorial](crosscompile_armv7.md).
```

Comment:
```suggestion
If you are interested in adding specific compiler flags to optimize operations
on your hardware, you can either set `CMAKE_CXX_FLAGS` manually, or look at the
copy of `CMake/crosscompile-arch-flags.cmake` in your project, find the
appropriate `ARCH_NAME` section, and add the new compilation flags to that file.
```
Hopefully I did not change the meaning of this too much; I tried to just clean it up and shorten it as much as possible.